examples/advanced/example_model_optimization.py

```diff
@@ -339,12 +341,14 @@ def suppress_output():
 
         with suppress_output():
             state_dict = hub.load_state_dict_from_url(url)
+        return state_dict
+
+    else:
+        raise RuntimeError("Please specify valid path to checkpoint")
 
-    transformer.load_state_dict(state_dict)
-else:
-    dataset = None
-    transformer = train(transformer, dataset)
 
+if root:
```

Comment:
Could we split this block here, by moving the note from the block above as well as the definitions of `root` and `checkpoint` down here?

```suggestion


########################################################################################
# .. note::
#
#   If you want to perform the training yourself, set ``root`` to the location
#   of the images downloaded from the link below

if root:
```